### PR TITLE
vmm: remove superfluous vcpu count

### PR DESCRIPTION
**Pull request by dianpopa**
## Reason for This PR

Remove superfluous vcpu count variable.

## Description of Changes

We can use the length of the vcpus variable
to get the count of vcpus. This way we also
discarding unused variables for x86_64.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria. Where there are two options, keep one.]`  
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [x] The reason for this PR is clearly provided. 
- [x] The description of changes is clear and encompassing.
- [x] No docs need to be updated as part of this PR.
- [x] Code-level documentation for touched code is included in this PR.
- [x] No API changes are included in this PR.
- [x] The changes in this PR have no user impact.
- [x] No new `unsafe` code has been added.

### CODE REVIEW

File: vmm/src/lib.rs [hunk 1]
```diff
@@ -1020,16 +1020,18 @@ impl Vmm {
...
-        // having set the vcpu count.
-        let vcpu_count = self.vm_config.vcpu_count.ok_or(VcpusNotConfigured)?;
-
         #[cfg(target_arch = "x86_64")]
         arch::x86_64::configure_system(
             vm_memory,
             GuestAddress(arch::x86_64::layout::CMDLINE_START),
             kernel_config.cmdline.len() + 1,
-            vcpu_count,
+            // If the length of the vcpus cannot be turned into an `u8`,
```
Comment (sandreim) [inline vmm/src/lib.rs]: The vcpu count is validated in machine_config.rs/validate_vcpu_num  against MAX_SUPPORTED_VCPUS and I am not sure if we actually need to validate again vs 256.

Comment (sandreim) [reply vmm/src/lib.rs]: You could just pass the vcpus.len() as u8 or into().

**Comment by acatangiu** [comment]
👍

File: vmm/src/lib.rs (hunk 1 above)
Comment (dianpopa) [reply vmm/src/lib.rs]: > The vcpu count is validated in machine_config.rs/validate_vcpu_num against MAX_SUPPORTED_VCPUS and I am not sure if we actually need to validate again vs u8 max value.

Here I am actually using the MAX_SUPPORTED_VCPUS as an argument for why the expect is safe. I am not trying to re-validate against MAX_SUPPORTED_VCPUS here.

Comment (dianpopa) [reply vmm/src/lib.rs]: > You could just pass the vcpus.len() as u8 or into().

Indeed `try_into` complicates things a bit. I used `as u8`. Take a look.

**Pull request merged**